core/src/main/scala/zio/logging/ContextMap.scala

```diff
@@ -2,7 +2,21 @@ package zio.logging
 
 import zio._
 
-final class ContextMap private (private val map: FiberRef[ContextMap.CMap]) {
+trait LoggingContext {
```

Comment:
This was pulled out merely to support SLF4j. It's making the whole interface more complicated only to support one particular implementation.

IMO I don't think that's a good design move. Either logging Context is a good idea in which case we should emulate it for SLF4J (which is relatively straightforward), or it's not a good idea, in which case we should remove it.

Reply:
This was not pulled to support SLF4j.... this was propose by your initial design. The only different is that you propose to tight interface to ```FiberRef[Map[String, Any]]``` and this is just a wrapper for that with context key that you propose to implement based on zio tests.